determined-ai/determined
# chore: Moving active experiment and task queries to separate stores [WEB-581] [WEB-582]

## Description

As part of breaking parts out of the central Store, this covers the active experiment and active tasks counts. These are used only on the Clusters page, but the experiment store should be OK for future queries.

Results of experiment queries are stored in the Immutable Map with the key generated by `encodeArgs`. This function uses an `any` type so we could put it in utils or shared or something for other stores.

Notes:
- Moves the code which polls the API from `pages/Clusters/ClustersOverview.tsx` to the same file which uses the data from the store `pages/Cluster/ClusterOverallStats.tsx`
- Why limit=-2 ? In `proto/src/determined/api/v1/experiment.proto` GetExperimentsRequest clarifies that limit=0 is the default page, limit=-1 is all results, so -2 is the first option which returns `pagination.total` without any experiments.

## Test Plan

- Visit `/det/clusters` - there should be a section showing active experiments, tensorboards, shells, and notebooks.
- Run an experiment and verify the count of experiments increases within a few seconds
- Use the Launch Jupyterlab button in the left sidebar, or `det shell start` to start a new task. Verify the count increases within a few seconds.
- Use the browser's dev console / Network tab to check API requests to `/api/v1/experiments?limit=-2` and `/api/v1/tasks/count` each second

## Checklist
See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
